uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-api/src/main/java/org/uberfire/ext/layout/editor/api/editor/LayoutComponent.java

```diff
@@ -39,52 +48,101 @@ public String getDragTypeName() {
         return dragTypeName;
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (!(o instanceof LayoutComponent)) {
-            return false;
-        }
-
-        LayoutComponent that = (LayoutComponent) o;
-
-        if (dragTypeName != null ? !dragTypeName.equals(that.dragTypeName) : that.dragTypeName != null) {
-            return false;
-        }
-        return !(properties != null ? !properties.equals(that.properties) : that.properties != null);
-    }
-
-    @Override
-    public int hashCode() {
-        int result = dragTypeName != null ? dragTypeName.hashCode() : 0;
-        result = 31 * result + (properties != null ? properties.hashCode() : 0);
-        return result;
-    }
-
     public Map<String, String> getProperties() {
         return properties;
     }
-
+    
     public void addProperty(String key,
                             String value) {
         properties.put(key,
                        value);
     }
+    
+    public void addPartProperty(String partId, 
+                                String key,
+                                String value) {
+        parts.stream().filter(p -> p.getPartId().equals(partId))
+                      .findFirst()
+                      .ifPresent(part -> part.addCssProperty(key, value));
+    }
 
     public void addProperties(Map<String, String> properties) {
-        for (String key : properties.keySet()) {
-            this.properties.put(key,
-                                properties.get(key));
+        properties.forEach(this.properties::put);
+    }
+    
+    public void addPartProperties(String partId, Map<String, String> properties) {
+        parts.stream().filter(p -> p.getPartId().equals(partId))
+                      .findFirst()
+                      .ifPresent(part -> properties.forEach(part::addCssProperty));
+    }
+    
+    public void addPartIfAbsent(String partId) {
+        Optional<LayoutComponentPart> containsPart = parts.stream().filter(p -> p.getPartId()
+                                                                   .equals(partId)).findFirst();
+        if (!containsPart.isPresent()) {
+            parts.add(new LayoutComponentPart(partId));
+        }
+    }
+    
+    public void removePartIf(Predicate<String> condition) {
+        parts.removeIf(p -> condition.test(p.getPartId()));
+    }
+    
+    public List<LayoutComponentPart> getParts() {
+        return Collections.unmodifiableList(parts);
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((dragTypeName == null) ? 0 : dragTypeName.hashCode());
```

Comment:
If this method is called in client-side, you have to do `result = ~~result` between each result attribution due to a GWT limitation.

Reply:
+1